### PR TITLE
Homework8

### PR DESCRIPTION
**Pull request by olyushIA**

### CODE REVIEW

File: HW8.html [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+<h3> 11. Знайти суму його парних дільників. </h3>
+<p id="11"></p>
+
+<h3> 12. Надрукувати повну таблицю множення від 1 до 10. </h3>
+<p id="12"></p>
+
+<script>
+    //Вивести на сторінку в один рядок через кому числа від 10 до 20.
+    let text1 = " ";
```
Comment (vitalii-prkh) [inline HW8.html]: To isolate each subtask you can use the IIFE function or block
```
// Block
{
 // here is the subtask
}

// IIFE
(() => {
 // here is the subtask
})()
```

File: HW8.html [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    let a = 10;
+    while (a < 21) {
```
Comment (vitalii-prkh) [inline HW8.html]: `while (a < 21)` or `while (a <= 20)`

File: HW8.html [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+        if (a != 20) {
+            text1 += a + ", ";
+            a++;
+        } else {
+            text1 += a;
+            a++;
+        }
+    }
+    document.getElementById("1").innerHTML = text1;
```
Comment (vitalii-prkh) [inline HW8.html]: How do you like such a solution?
```
let a = 10;
const result = [];

while (a <= 20) {
  result.push(a);
  a++;
}

document.getElementById("1").innerHTML = t = result.join(", ");
```

File: HW8.html (hunk 2 above)
Comment (vitalii-prkh) [reply HW8.html]:
```
if (a != 20) {
  text1 += a + ", ";
} else {
  text1 += a;
}

a++; // increment after if/else statement
```

File: HW8.html [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+        }
+    }
+    document.getElementById("1").innerHTML = text1;  
+    
+    //Вивести квадрати чисел від 10 до 20.
+    let text2 = " ";
+    let b = 10;
+    while (b < 21) {
+        if (b != 20) {
+            text2 += Math.pow(b, 2) + ", ";
```
Comment (vitalii-prkh) [inline HW8.html]: Take into account previous comment with .join(", ") and refactor

File: HW8.html [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+            b++;
+        } else {
+            text2 += Math.pow(b, 2);
+            b++;
+        }
+    }
+    document.getElementById("2").innerHTML = text2;
+
+    //Вивести таблицю множення на 7.
+    let text3 = "";
```
Comment (vitalii-prkh) [inline HW8.html]:
```
let c = 1;
const list = [];

while (c <= 10) {
  list.push(`${c} * 7 = ${c * 7}`);
  c++;
}

document.getElementById("3").innerHTML = list.join("<br />");
```

Comment (vitalii-prkh) [reply HW8.html]:
```
// More semantic, with HTML list tags
let c = 1;
const list = [];

while (c <= 10) {
  list.push(`<li>${c} * 7 = ${c * 7}</li>`);
  c++;
}

document.getElementById("3").innerHTML = `<ul>${list.join("")}</ul>`;
```

File: HW8.html [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+        f++;
+    } while (f < 501);
+    mean = mean / 500;
+    document.getElementById("6").innerHTML = mean;
+
+    //Вивести суму лише парних чисел в діапазоні від 30 до 80.
+    let evenSum = 0;
+    for (let g = 30; g < 81; g++) {
+        if (g % 2 == 0) {
+            evenSum = evenSum + +g;
```
Comment (vitalii-prkh) [inline HW8.html]: Finalize, and fix this unnecessary plus `+g` in all the tasks
I showed you how to isolate each subtask

File: HW8.html (hunk 1 above)
Comment (olyushIA) [reply HW8.html]: Updated

File: HW8.html (hunk 3 above)
Comment (olyushIA) [reply HW8.html]: Looks great, I like it!
Why do you add = t = to the last line?

File: HW8.html (hunk 5 above)
Comment (olyushIA) [reply HW8.html]: I'll use simpler version above, thanks!

File: HW8.html [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+        const list = [];
+        while (c <= 10) {
+            list.push(`${c} * 7 = ${c * 7}`);
+            c++;
+            }
+        document.getElementById("3").innerHTML = list.join("<br />");
+    }
+
+    //Знайти суму всіх цілих чисел від 1 до 15.
+    {   let text4 = "";
```
Comment (vitalii-prkh) [inline HW8.html:104]:
```
const numbers = [];
let sum = 0;
let d = 1;

do {
  numbers.push(d);
  d++;
} while (d <= 15);

document.getElementById("4").innerHTML = `${numbers.join(" + ")} = ${numbers.reduce((x, y) => x + y, 0)}`;
```

Comment (vitalii-prkh) [reply HW8.html:104]: One more solution with array methods

```
const numbers = Array.from({ length: 15 }, (_, i) => i + 1);

console.log(`${numbers.join(" + ")} = ${numbers.reduce((x, y) => x + y, 0)}`);
```

File: HW8.html [hunk 8]
```diff
@@ -0,0 +1,229 @@
...
+        text4 = textA + " = " + sum;
+        document.getElementById("4").innerHTML = text4;
+    }
+
+    //Знайти добуток усіх цілих чисел від 15 до 35.
+    {   let text5 = "";
+        let textB = "";
+        let mult = 1;
+        let e = 15;
+        do {
```
Comment (vitalii-prkh) [inline HW8.html:126]: Read previous comments

File: HW8.html [hunk 9]
```diff
@@ -0,0 +1,229 @@
...
+            }
+            mult = mult * e;
+            e++;
+        } while (e < 36);
+        text5 = textB + " = " + BigInt(mult);
+        document.getElementById("5").innerHTML = text5;
+    }
+
+    //Знайти середнє арифметичне всіх цілих чисел від 1 до 500.
+    {   let mean = 0;
```
Comment (vitalii-prkh) [inline HW8.html:140]:
```
let sum = 0;
let f = 1;

do {
  sum = sum + f;
  f++;
} while (f <= 500);

document.getElementById("6").innerHTML = sum / 500;
```

File: HW8.html (hunk 8 above)
Comment (olyushIA) [reply HW8.html:126]: Thanks, I'll take into consideration

**Pull request merged**